app/views/handlers/block_factory.py

```diff
@@ -31,16 +32,43 @@
 }
 
 
-def get_block_handler(schema, location, questionnaire_store, language):
-    block = schema.get_block(location.block_id)
+def get_block_handler(
+    schema, block_id, list_item_id, questionnaire_store, language, **kwargs
+):
+    block = schema.get_block(block_id)
     if not block:
         raise InvalidLocationException(
-            f'block id {location.block_id} is not valid for this schema'
+            f'block id {block_id} is not valid for this schema'
+        )
+
+    if schema.is_block_in_repeating_section(block_id=block['id']) and not all(
+        (kwargs.get('list_name'), list_item_id)
+    ):
+        raise InvalidLocationException(
+            f'block id {block_id} is in a repeating section without valid list_name/list_item_id'
         )
 
     block_type = block['type']
     block_class = BLOCK_MAPPINGS.get(block_type)
     if not block_class:
         raise ValueError(f'block type {block_type} is not valid')
 
+    section_id = schema.get_section_id_for_block_id(block_id)
+    to_list_item_id = kwargs.get('to_list_item_id')
```

Comment:
Can we make this a try except block using KeyError for the dict access?

Reply:
Same as my previous comments.